src/delta.rs

```diff
@@ -114,12 +114,27 @@ where
                 handle_hunk_meta_line(&mut painter, &line, config)?;
                 continue;
             }
-        } else if source == Source::DiffUnified && line.starts_with("Only in ") {
+        } else if source == Source::DiffUnified && line.starts_with("Only in ")
+            || line.starts_with("Submodule ")
+        {
+            // Additional FileMeta cases:
+            //
+            // 1. When comparing directories with diff -u, if filenames match between the
+            //    directories, the files themselves will be compared. However, if an equivalent
+            //    filename is not present, diff outputs a single line (Only in...) starting
+            //    indicating that the file is present in only one of the directories.
+            //
+            // 2. Git diff emits lines describing submodule state such as "Submodule x/y/z contains
+            //    untracked content"
+            //
+            // See https://github.com/dandavison/delta/issues/60#issuecomment-557485242 for a
+            // proposal for more robust parsing logic.
```

Comment:
@fdcds your implementation proposal in #60 sounds much more robust than what I've done here. I didn't feel quite ready to go ahead with it yet so have gone with this "easy option". This state machine part of the delta codebase is really hard to understand and fragile, as I expect @m-lima will agree with. I think it would make sense to investigate better abstractions for implementing the parser here.

Reply:
/cc @devurandom